packages/cursorless-engine/src/processTargets/targets/UntypedTarget.ts

```diff
@@ -1,12 +1,38 @@
 import { Range } from "@cursorless/common";
-import { BaseTarget, CommonTargetParameters } from ".";
+import { BaseTarget, MinimumTargetParameters, CommonTargetParameters } from ".";
 import type { Target } from "../../typings/target.types";
-import { createContinuousRangeUntypedTarget } from "../targetUtil/createContinuousRange";
 import {
   getTokenLeadingDelimiterTarget,
   getTokenRemovalRange,
   getTokenTrailingDelimiterTarget,
 } from "../targetUtil/insertionRemovalBehaviors/TokenInsertionRemovalBehavior";
+import { createContinuousRange } from "../targetUtil/createContinuousRange";
+import { isSameType } from "../../util/typeUtils";
+
+/**
+ * An abstract target base class who createContinueRangeTarget method returns either
+ * a createContinuousRange if this and endTarget have the same type or
+ * otherwise returns an UntypedTarget from this to endTarget.
+ */
+export abstract class CommonTarget<
```

Comment:
I think I may call this `CommonBaseTarget`? Idk wdyt @auscompgeek

Reply:
`CommonBase` sounds like a bit of a tautology to me 😄

Reply:
What about `CommonContinuousRangeTarget`?  Any name is fine with me. :-)